server/api/api/endpoints/functions.py

```diff
@@ -185,6 +186,23 @@ async def delete_function(
             await run_in_threadpool(
                 get_scheduler().delete_schedule, db_session, project, name
             )
+
+    fn = await run_in_threadpool(
+        server.api.crud.Functions().get_function, db_session, name, project
+    )
+    if fn.get("kind") in RuntimeKinds.pure_nuclio_deployed_runtimes():
```

Comment:
Should apply for application runtime as well
```suggestion
    if fn.get("kind") in RuntimeKinds.nuclio_runtimes():
```